homeassistant/components/qnap_qsw/__init__.py

```diff
@@ -0,0 +1,89 @@
+"""The QNAP QSW integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from aioqsw.const import (
+    QSD_FIRMWARE,
+    QSD_FIRMWARE_INFO,
+    QSD_MAC,
+    QSD_PRODUCT,
+    QSD_SYSTEM_BOARD,
+)
+from aioqsw.localapi import ConnectionOptions, QnapQswApi
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import QswUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+class QswEntity(CoordinatorEntity[QswUpdateCoordinator]):
+    """Define an QNAP QSW entity."""
+
+    def __init__(
+        self,
+        coordinator: QswUpdateCoordinator,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator)
+
+        self._attr_device_info = DeviceInfo(
+            configuration_url=entry.data[CONF_URL],
+            connections={
+                (
+                    CONNECTION_NETWORK_MAC,
+                    self.get_entity_value(QSD_SYSTEM_BOARD, QSD_MAC),
+                )
+            },
+            manufacturer=MANUFACTURER,
+            model=self.get_entity_value(QSD_SYSTEM_BOARD, QSD_PRODUCT),
+            name=self.get_entity_value(QSD_SYSTEM_BOARD, QSD_PRODUCT),
+            sw_version=self.get_entity_value(QSD_FIRMWARE_INFO, QSD_FIRMWARE),
+        )
+
+    def get_entity_value(self, key: str, subkey: str) -> Any:
```

Comment:
Should this be called `get_zone_value`?  Its a bit confusing that it has the name entity in it

Reply:
Maybe `get_switch_value` or `get_device_value`?

Reply:
`get_device_value` seems better to me.  Naming is hard though.

Please be sure to adjust the doc string

Reply:
Renamed in c8cb7495c7842be6b61fece97106ae894b7ae6ce